cartoframes/dataset.py

```diff
@@ -55,52 +58,61 @@ def __init__(self, table_name=None, schema='public', query=None, df=None, gdf=No
             self.normalized_column_names = _normalize_column_names(gdf)
 
     @classmethod
-    def from_table(cls, table_name, schema='public', context=None):
-        return cls(
-            table_name=table_name, schema=schema, context=context, state=cls.STATE_REMOTE)
+    def from_table(cls, table_name, context, schema='public'):
+        return cls(table_name=table_name, schema=schema, context=context, state=cls.STATE_REMOTE)
 
     @classmethod
-    def from_query(cls, query, context=None):
-        return cls(
-            query=query, context=context, state=cls.STATE_REMOTE)
+    def from_query(cls, query, context):
+        return cls(query=query, context=context, state=cls.STATE_REMOTE)
 
     @classmethod
-    def from_dataframe(cls, df, table_name=None, schema='public', context=None):
-        return cls(
-            df=df, table_name=table_name, schema=schema, context=context, state=cls.STATE_LOCAL)
+    def from_dataframe(cls, df):
+        return cls(df=df, state=cls.STATE_LOCAL)
 
     @classmethod
-    def from_geodataframe(cls, gdf, table_name=None, schema='public', context=None):
-        return cls(
-            gdf=gdf, table_name=table_name, schema=schema, context=context, state=cls.STATE_LOCAL)
+    def from_geodataframe(cls, gdf):
+        return cls(gdf=gdf, state=cls.STATE_LOCAL)
 
     @classmethod
-    def from_geojson(cls, geojson, table_name=None, schema='public', context=None):
-        return cls(
-            gdf=load_geojson(geojson), table_name=table_name, schema=schema, context=context, state=cls.STATE_LOCAL)
+    def from_geojson(cls, geojson):
+        return cls(gdf=load_geojson(geojson), state=cls.STATE_LOCAL)
 
-    def upload(self, with_lonlat=None, if_exists='fail'):
-        if self.query and self.table_name is not None and not self.exists():
-            self.cc.batch_sql_client.create_and_wait_for_completion(
-                '''BEGIN; {drop}; {create}; {cartodbfy}; COMMIT;'''
-                .format(drop=self._drop_table_query(),
-                        create=self._create_table_from_query(self.query),
-                        cartodbfy=self._cartodbfy_query()))
-        else:
-            if self.df is None:
-                raise ValueError('You have to create a `Dataset` with a pandas.DataFrame to upload it to CARTO')
+    def upload(self, with_lonlat=None, if_exists='fail', table_name=None, schema=None, context=None):
```

Comment:
I'm not sure in which cases we need to send the schema...

Reply:
I've only seen the need for it in reading datasets (e.g., if someone shares a dataset with you on an org account). I can't think of a case where we need this for uploading datasets.

Reply:
Appending data to a shared table